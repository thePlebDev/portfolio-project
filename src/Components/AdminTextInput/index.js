import React from 'react'
import styled from 'styled-components';

const Container = styled.div`
  margin-top:10px;
  margin-bottom:10px;
`
const Input = styled.input`
  width:60%;
  padding:10px;
  border:${props=>props.state ? '1px solid red':'1px solid black'};
`

const AdminTextInput = ({state="enter state prop",handleChange="enter handleChange prop",error="enter error prop",name="enter name prop"})=>{

  return(
    <Container data-testid="container">
      <label for='username'>
        <Input id='username' data-testid="input" state={error} name={name} type="text" value={state} onChange={(e)=>handleChange(e)} placeholder={name}/>
      </label>
    </Container>
  )
}

export default AdminTextInput
