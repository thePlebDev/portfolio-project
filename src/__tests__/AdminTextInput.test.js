import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import AdminTextInput from '../Components/AdminTextInput';


//1) (DONE)I need to test the snapshot
//2) (DONE) I need to test that all the major parts render properly
//3) how does it handle no props
//3) how does it handle custom props
//4) simulate event. For forms its a change event
//5) make sure the handleChange function gets called.

describe('testing the AdminTextInput',()=>{
  it('should test the snapshot function',()=>{
    const tree = renderer.create(<AdminTextInput/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should render everything properly',()=>{
    const wrapper = shallow(<AdminTextInput/>)
    const container = wrapper.find('[data-testid="container"]')
    const input = wrapper.find('[data-testid="input"]')
    expect(container.length).toBe(1)
    expect(input.length).toBe(1)
  })

  it('should deal with no props accordingly',()=>{
    const wrapper = shallow(<AdminTextInput/>)
    const input = wrapper.find('[data-testid="input"]')
    expect(input.props().name).toBe('enter name prop')
    expect(input.props().state).toBe('enter error prop')
    expect(input.props().value).toBe('enter state prop')
    expect( typeof input.props().onChange).toBe('function')
  })

  it('should deal with custome props accordingly',()=>{
    const mockHandleChange = jest.fn()
    const wrapper = shallow(<AdminTextInput state={'state'} error={'error'} handleChange={mockHandleChange} name={'name'} />)
    const input = wrapper.find('[data-testid="input"]')
    expect(input.props().name).toBe('name')
    expect(input.props().state).toBe('error')
    expect(input.props().value).toBe('state')
    expect( typeof input.props().onChange).toBe('function')
  })

  it('should test the event change',()=>{
    const mockHandleChange = jest.fn()
    const wrapper = shallow(<AdminTextInput state={'state'} error={'error'} handleChange={mockHandleChange} name={'name'} />)
    const input = wrapper.find('[data-testid="input"]')
    input.simulate('change',{
      target:{name:'password',value:'Jack'},
    })
    expect(mockHandleChange.mock.calls.length).toBe(1)
  })
})
