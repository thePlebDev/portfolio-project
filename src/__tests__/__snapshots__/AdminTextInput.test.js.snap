// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the AdminTextInput should test the snapshot function 1`] = `
<div
  className="sc-bdnylx cKBbeu"
  data-testid="container"
>
  <label
    for="username"
  >
    <input
      className="sc-gtssRu fOFZCV"
      data-testid="input"
      id="username"
      name="enter name prop"
      onChange={[Function]}
      placeholder="enter name prop"
      type="text"
      value="enter state prop"
    />
  </label>
</div>
`;
